### PR TITLE
feat: warning if backdrop click or escape key is pressed when new action form is dirty

### LINKED ISSUE #5681: Warn about unsaved changes when editing Actions
As a dev, I want to get a warning when I close/navigate away from editing Actions so that I don't loose my code.

Closes Action Modal without warning:
- Accidentally navigating back (swipe gesture)
- Clicking anywhere outside the modal

```[tasklist]
### Acceptance Criteria
- [ ] Warn users before closing the modal with unsaved changes
```

### PR DESCRIPTION
**Pull request by doncicuto**
As a dev, I want to get a warning when I click anywhere outside the modal or the escape key is pressed

Acceptance Criteria

- [X] Warn users before closing the modal with unsaved changes. 

Deviation from acceptance criteria:

- I tried to tackle the "Accidentally navigating back (swipe gesture)" issue using router outlets, auxiliary routes and route guards for dialogs (new action and update action). Angular/Material Dialog didn't help me to find an elegant way to show the warning message, code was too convoluted so I prefer to stash files.

This is a sample video showing backdrop click or when escape key is pressed:

https://github.com/zitadel/zitadel/assets/30386061/c72b8f24-7ff2-4ec1-a404-a7149d1f29c0

Would close #5681 

### Definition of Ready

- [X] I am happy with the code
- [X] Short description of the feature/issue is added in the pr description
- [X] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [X] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [X] No debug or dead code
- [ ] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [X] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Comment by hifabienne** [comment]
hei @doncicuto thanks for your work.
We will have a look at it.

**Comment by hifabienne** [comment]
@peintnermax can you have a look at this?

File: console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts [hunk 1]
```diff
@@ -26,12 +29,15 @@ export class AddActionDialogComponent {
   public id: string = '';
 
   public opened$ = this.dialogRef.afterOpened().pipe(mapTo(true));
-
+  public form!: UntypedFormGroup;
```
Comment (peintnermax) [inline console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts:32]:
```suggestion
  public form: FormGroup = new FormGroup({
    name: new FormControl<string>('', []),
    script: new FormControl<string>('', []),
    durationInSec: new FormControl<number>(10, []),
    allowedToFail: new FormControl<boolean>(false, []),
  });
```
can we initialize this with typedforms and remove the fixed variables on top?

File: console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts [hunk 2]
```diff
@@ -17,7 +20,7 @@ import { ToastService } from 'src/app/services/toast.service';
   templateUrl: './add-action-dialog.component.html',
   styleUrls: ['./add-action-dialog.component.scss'],
 })
-export class AddActionDialogComponent {
+export class AddActionDialogComponent implements OnInit {
   public name: string = '';
   public script: string = '';
   public durationInSec: number = 10;
```
Comment (peintnermax) [inline console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts:26]:
```suggestion
```
not needed anymore if we initialize the form directly and overwrite if anydata is set

File: console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts [hunk 3]
```diff
@@ -44,6 +50,53 @@ export class AddActionDialogComponent {
       this.allowedToFail = action.allowedToFail;
```
Comment (peintnermax) [inline console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts:50]:
```suggestion
      this.allowedToFail = action.allowedToFail;
```
```suggestion
      this.form.setValue({
        name: action.name,
        script: action.script,
        durationInSec: action.timeout?.seconds ?? 10,
        allowedToFail: action.allowedToFail,
      });
```

**Review by peintnermax — CHANGES_REQUESTED** [review]
Thank you for your effort.. 🙏 I've made a suggestion to use typed forms and initialize it on top. What do you think?

File: console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts [hunk 4]
```diff
@@ -53,28 +106,28 @@ export class AddActionDialogComponent {
   public closeDialogWithSuccess(): void {
     if (this.id) {
       const req = new UpdateActionRequest();
-      req.setId(this.id);
-      req.setName(this.name);
-      req.setScript(this.script);
+      req.setId(this.form.value.id);
```
Comment (peintnermax) [inline console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts:109]:
```suggestion
      req.setId(this.id);
```

This won't work as this value is empty.
We have to set the id directly.

**Comment by doncicuto** [comment]
Thanks a lot @peintnermax, as always, for your review. I'll submit a PR including your suggested changes.

File: console/src/assets/i18n/de.json [hunk 5]
```diff
@@ -166,7 +166,15 @@
     "NEXT": "Weiter",
     "MORE": "mehr",
     "STEP": "Schritt",
-    "COMINGSOON": "Coming soon",
+    "COMINGSOON": "Demnächst",
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/assets/i18n/de.json]: Changed! :smile:

File: console/src/assets/i18n/en.json [hunk 6]
```diff
@@ -168,6 +168,14 @@
     "STEP": "Step",
     "SETUP": "Setup",
     "COMINGSOON": "Coming soon",
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/assets/i18n/en.json]: Changed! :smile:

File: console/src/assets/i18n/es.json [hunk 7]
```diff
@@ -168,6 +168,14 @@
     "STEP": "Paso",
     "SETUP": "Configurar",
     "COMINGSOON": "Próximamente",
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/assets/i18n/es.json]: Changed! :smile:

File: console/src/assets/i18n/fr.json [hunk 8]
```diff
@@ -166,7 +166,15 @@
     "NEXT": "Suivant",
     "MORE": "plus",
     "STEP": "Étape",
-    "COMINGSOON": "Coming soon",
+    "COMINGSOON": "À venir",
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/assets/i18n/fr.json]: Changed! :smile:

File: console/src/assets/i18n/pl.json [hunk 9]
```diff
@@ -166,7 +166,15 @@
     "NEXT": "Następny",
     "MORE": "więcej",
     "STEP": "Krok",
-    "COMINGSOON": "Coming soon",
+    "COMINGSOON": "Wkrótce",
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/assets/i18n/pl.json]: Changed! :smile:

File: console/src/assets/i18n/zh.json [hunk 10]
```diff
@@ -165,8 +165,16 @@
     "PREVIOUS": "上一页",
     "NEXT": "下一页",
     "MORE": "更多",
-    "STEP": "Step",
-    "COMINGSOON": "Coming soon",
+    "STEP": "步",
+    "COMINGSOON": "即将推出",
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/assets/i18n/zh.json]: Changed! :smile:

File: console/src/assets/i18n/it.json [hunk 11]
```diff
@@ -166,7 +166,14 @@
     "NEXT": "Avanti",
     "MORE": "azioni",
     "STEP": "Passo",
-    "COMINGSOON": "Coming soon",
+    "COMINGSOON": "Prossimamente",
+    "UNSAVED": {
+      "DIALOG": {
+        "DESCRIPTION": "Sei sicuro di voler eliminare questa nuova azione? La tua azione andrà persa",
+        "CANCEL": "Cancella",
+        "DISCARD": "Scartare"
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/assets/i18n/it.json]: Changed! :smile:

File: console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts (hunk 1 above)
Comment (doncicuto) [reply console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts]: Absolutely!

File: console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts (hunk 3 above)
Comment (doncicuto) [reply console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts]: Changed! :smile:

File: console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts (hunk 4 above)
Comment (doncicuto) [reply console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts]: Fixed, thanks a lot

File: console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts (hunk 2 above)
Comment (doncicuto) [reply console/src/app/pages/actions/add-action-dialog/add-action-dialog.component.ts]: Code removed!

File: console/src/assets/i18n/ja.json [hunk 12]
```diff
@@ -168,6 +168,14 @@
     "STEP": "ステップ",
     "SETUP": "セットアップ",
     "COMINGSOON": "近日公開",
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/assets/i18n/ja.json]: Changed! :smile:

File: console/src/assets/i18n/it.json [hunk 13]
```diff
@@ -166,7 +166,14 @@
     "NEXT": "Avanti",
     "MORE": "azioni",
     "STEP": "Passo",
-    "COMINGSOON": "Coming soon",
+    "COMINGSOON": "Prossimamente",
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/assets/i18n/it.json]: Changed! :smile:

**Review by peintnermax — APPROVED** [review]

**Pull request merged**